src/utils/errors.rs

```diff
@@ -356,3 +268,16 @@ impl std::fmt::Display for WrappedResponseContent {
         write!(f, "status: {} content: {}", self.status, self.content)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::PineconeError;
+    use tokio;
+
+    fn assert_send_sync<T: Send + Sync>() {}
+
+    #[tokio::test]
+    async fn test_pinecone_error_is_send_sync() {
+        assert_send_sync::<PineconeError>();
+    }
```

Comment:
Testing to ensure we're properly implementing `Send + Sync` on the custom error type.